File: code-generator/src/main/java/io/github/potjerodekool/openapi/internal/generate/api/RequestGenerator.java
package io.github.potjerodekool.openapi.internal.generate.api;

import io.github.potjerodekool.codegen.Environment;
import io.github.potjerodekool.codegen.io.Filer;
import io.github.potjerodekool.codegen.io.Location;
import io.github.potjerodekool.openapi.GeneratorConfig;
import io.github.potjerodekool.openapi.internal.generate.Templates;

import java.io.IOException;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class RequestGenerator {

    private final String basePackageName;
    private final Filer filer;
    private final Templates templates;

    public RequestGenerator(final GeneratorConfig generatorConfig,
                            final Environment environment,
                            final Templates templates) {
        this.basePackageName = generatorConfig.basePackageName();
        this.filer = environment.getFiler();
        this.templates = templates;
    }

    public void generate() {
        final var st = templates.getInstanceOf("/request/request");

        st.add("packageName", this.basePackageName);
        st.add("generatorName", getClass().getName());
        st.add("date", DateTimeFormatter.ISO_DATE_TIME.format(LocalDateTime.now()));

        final var code = st.render();

        final var resource = filer.createResource(
                Location.SOURCE_OUTPUT,
                this.basePackageName,
                "Request.java"
        );

        resource.writeToOutputStream(code.getBytes());
    }
}

File: generator-common/src/main/kotlin/io/github/potjerodekool/openapi/common/generate/api/RequestGenerator.kt
package io.github.potjerodekool.openapi.common.generate.api

import io.github.potjerodekool.codegen.Environment
import io.github.potjerodekool.codegen.io.Filer
import io.github.potjerodekool.codegen.io.Location
import io.github.potjerodekool.openapi.common.GeneratorConfig
import io.github.potjerodekool.openapi.common.generate.Templates
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

class RequestGenerator(
    generatorConfig: GeneratorConfig,
    environment: Environment,
    private val templates: Templates
) {
    private val basePackageName = generatorConfig.basePackageName
    private val filer: Filer = environment.filer

    fun generate() {
        val st = templates.getInstanceOf("/request/request")

        st.add("packageName", this.basePackageName)
        st.add("generatorName", javaClass.name)
        st.add("date", DateTimeFormatter.ISO_DATE_TIME.format(LocalDateTime.now()))

        val code = st.render()

        val resource = filer.createResource(
            Location.SOURCE_OUTPUT,
            this.basePackageName,
            "Request.java"
        )

        resource.writeToOutputStream(code.toByteArray())
    }
}
